src/ir/effects.h

```diff
@@ -589,6 +589,11 @@ class EffectAnalyzer {
     void visitRefIs(RefIs* curr) {}
     void visitRefFunc(RefFunc* curr) {}
     void visitRefEq(RefEq* curr) {}
+    void visitTableGet(TableGet* curr) {
+      // TODO: track readsTable/writesTable, like memory?
```

Comment:
Yep, tracking that sounds important. Want to model this as reading from memory for now? `call_indirect` should probably be modeled as reading from memory to match if so.

Reply:
Until we get `table.set` this doesn't seem like a potential problem, though? At that time, I think we should add `readsTable` etc. I suppose.

Reply:
Well `call_indirect` reads the table, but I guess in practice it doesn't matter since it also does a call. Not doing anything until we also have `table.set` sgtm, then.